src/colorizer/ColorizeCanvas3D.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  AREA_LIGHT,
+  Light,
+  LoadSpec,
+  Lut,
+  RawArrayLoader,
+  RENDERMODE_RAYMARCH,
+  SKY_LIGHT,
+  TiffLoader,
+  View3d,
+  Volume,
+  VolumeLoaderContext,
+  WorkerLoader,
+} from "@aics/vole-core";
+import { Vector2, Vector3 } from "three";
+
+import { CanvasScaleInfo, CanvasType, FrameLoadResult } from "./types";
+
+import { IRenderCanvas, RenderCanvasStateParams } from "./IRenderCanvas";
+
+const CACHE_MAX_SIZE = 1_000_000_000;
+const CONCURRENCY_LIMIT = 8;
+const PREFETCH_CONCURRENCY_LIMIT = 3;
+const loaderContext = new VolumeLoaderContext(CACHE_MAX_SIZE, CONCURRENCY_LIMIT, PREFETCH_CONCURRENCY_LIMIT);
+
+export class ColorizeCanvas3D implements IRenderCanvas {
+  // private viewContainer: HTMLElement;
+  private view3d: View3d;
+  private onLoadFrameCallback: (result: FrameLoadResult) => void;
+  // private params: RenderCanvasStateParams;
+
+  private canvasResolution: Vector2;
+
+  private tempCanvas: HTMLCanvasElement;
+
+  private loader: WorkerLoader | RawArrayLoader | TiffLoader | null = null;
+  private volume: Volume | null = null;
+  private pendingVolumePromise: Promise<FrameLoadResult> | null = null;
+  private pendingFrame: number;
+  private currentFrame: number;
+
+  constructor(_params: RenderCanvasStateParams) {
+    // this.params = params;
+
+    this.view3d = new View3d();
+    this.view3d.loaderContext = loaderContext;
+    this.canvasResolution = new Vector2(10, 10);
+    this.setResolution(10, 10);
+    this.view3d.setShowAxis(true);
+    this.view3d.setVolumeRenderMode(RENDERMODE_RAYMARCH);
+    this.initLights();
+
+    this.tempCanvas = document.createElement("canvas");
+    this.tempCanvas.style.width = "10px";
+    this.tempCanvas.style.height = "10px";
+    this.pendingFrame = -1;
+    this.currentFrame = -1;
+
+    this.onLoadFrameCallback = () => {};
+  }
+
+  private initLights(): void {
+    const lights = [new Light(SKY_LIGHT), new Light(AREA_LIGHT)];
+    lights[0].mColorTop = new Vector3(0.3, 0.3, 0.3);
+    lights[0].mColorMiddle = new Vector3(0.3, 0.3, 0.3);
+    lights[0].mColorBottom = new Vector3(0.3, 0.3, 0.3);
+    lights[1].mTheta = (14 * Math.PI) / 180.0;
+    lights[1].mPhi = (54 * Math.PI) / 180.0;
+    lights[1].mColor = new Vector3(0.3, 0.3, 0.3);
+    this.view3d.updateLights(lights);
```

Comment:
honestly I dont think the lights actually do anything unless you are displaying isosurfaces (which actually could be interesting for segmentations but less performant)

Reply:
I'll fully admit I added these while I was thrashing trying to figure out why volumes weren't rendering (it was because of an unrelated worker import issue).